types/stylelint/index.d.ts

```diff
@@ -315,28 +351,44 @@ declare namespace stylelint {
 				ignoreAtRules: OneOrMany<string>;
 			}
 		>;
-		'at-rule-no-deprecated': CoreRule<true, { ignoreAtRules: OneOrMany<StringOrRegex> }>;
-		'at-rule-no-unknown': CoreRule<true, { ignoreAtRules: OneOrMany<StringOrRegex> }>;
-		'at-rule-no-vendor-prefix': CoreRule<true>;
-		'at-rule-prelude-no-invalid': CoreRule<true, { ignoreAtRules: OneOrMany<StringOrRegex> }>;
-		'at-rule-property-required-list': CoreRule<Record<string, OneOrMany<string>>>;
+		'at-rule-no-deprecated': CoreRule<
+			true,
+			{ ignoreAtRules: OneOrMany<StringOrRegex> },
+			RejectedMessage<[atRule: string]>
+		>;
+		'at-rule-no-unknown': CoreRule<
+			true,
+			{ ignoreAtRules: OneOrMany<StringOrRegex> },
+			RejectedMessage<[atRule: string]>
+		>;
+		'at-rule-no-vendor-prefix': CoreRule<true, {}, RejectedMessage<[atRule: string]>>;
+		'at-rule-prelude-no-invalid': CoreRule<
+			true,
+			{ ignoreAtRules: OneOrMany<StringOrRegex> },
+			RejectedMessage<[atRule: string, prelude: string]>
+		>;
+		'at-rule-property-required-list': CoreRule<
+			Record<string, OneOrMany<string>>,
+			{},
+			ExpectedMessage<[atRule: string, property: string] | [atRule: string, descriptor: string]>
+		>;
 		'block-no-empty': CoreRule<true, { ignore: OneOrMany<'comments'> }>;
-		'color-function-notation': CoreRule<
+		'color-function-notation': NotationRule<
 			'modern' | 'legacy',
 			{ ignore: OneOrMany<'with-var-inside'> }
 		>;
-		'color-hex-alpha': CoreRule<'always' | 'never'>;
-		'color-hex-length': CoreRule<'short' | 'long', {}, AutofixMessage>;
+		'color-hex-alpha': CoreRule<
+			'always' | 'never',
+			{},
+			ExpectedMessage<[hex: string]> & RejectedMessage<[hex: string]>
+		>;
+		'color-hex-length': CoreRule<'short' | 'long', {}, ExpectedMessage<'autofix'>>;
 		'color-named': CoreRule<
 			'never' | 'always-where-possible',
-			{ ignoreProperties: OneOrMany<StringOrRegex>; ignore: OneOrMany<'inside-function'> },
-			{
-				expected: (expected: string, actual: string) => string;
-				rejected: (keyword: string) => string;
-			}
```

Comment:
`messageArgs` wasn't added to rules that require for the user to check the `arguments` length.
e.g.
```
message: (a, b) {
  // or arguments.length === 2
  if (typeof b === 'string') {
    /* … */
  } else { /* … */ }
}
```

It may be added so that #6966 can be resolved but that's out of scope for this PR.